### PR TITLE
[Refactoring] More self described test in MessageFastViewProjectionContract

### PR DESCRIPTION
**Pull request by trantienduchn**
By less mathematics involved

References: https://github.com/linagora/james-project/pull/2913#discussion_r353096178

### CODE REVIEW

**Comment by trantienduchn** [comment]
```
[ad52942008d323e3c54c3c23d309e6b527b58ce0] [INFO] Running org.apache.james.queue.file.FileMailQueueFactoryTest
[ad52942008d323e3c54c3c23d309e6b527b58ce0] [INFO] Tests run: 8, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.741 s - in org.apache.james.queue.file.FileMailQueueFactoryTest
[ad52942008d323e3c54c3c23d309e6b527b58ce0] [INFO] Running org.apache.james.queue.file.FileMailQueueTest
[ad52942008d323e3c54c3c23d309e6b527b58ce0] [ERROR] Tests run: 73, Failures: 0, Errors: 1, Skipped: 9, Time elapsed: 71.778 s <<< FAILURE! - in org.apache.james.queue.file.FileMailQueueTest
[ad52942008d323e3c54c3c23d309e6b527b58ce0] [ERROR] dequeueShouldBeConcurrent  Time elapsed: 61.033 s  <<< ERROR!
[ad52942008d323e3c54c3c23d309e6b527b58ce0] org.awaitility.core.ConditionTimeoutException: Condition with lambda expression in org.apache.james.queue.api.MailQueueContract that uses java.util.concurrent.ConcurrentLinkedDeque, java.util.concurrent.ConcurrentLinkedDequeint was not fulfilled within 60 seconds.
```
test this please

**Comment by trantienduchn** [comment]
```
[ad52942008d323e3c54c3c23d309e6b527b58ce0] [ERROR] Errors: 
[ad52942008d323e3c54c3c23d309e6b527b58ce0] [ERROR]   EventDeadLettersIntegrationTest.failedEventShouldNotBeInDeadLettersAfterSuccessfulRedelivery:334->retrieveFirstFailedInsertionId:197 » IndexOutOfBounds
```
test this please

File: server/data/data-jmap/src/test/java/org/apache/james/jmap/api/projections/MessageFastViewProjectionContract.java [hunk 1]
```diff
@@ -236,25 +236,24 @@ default void concurrentStoreShouldOverrideOldValueWhenSameMessageId() throws Exc
     @Test
     default void storeShouldBeConsistentUponSingleKeyOperation() throws Exception {
```
Comment (mbaechler) [inline server/data/data-jmap/src/test/java/org/apache/james/jmap/api/projections/MessageFastViewProjectionContract.java]: storeShouldReturnAnyLatestPreviewOnConcurrentUpdate

Comment (chibenwa) [reply server/data/data-jmap/src/test/java/org/apache/james/jmap/api/projections/MessageFastViewProjectionContract.java]: Applied upon merge

**Comment by chibenwa** [comment]
Merged

**Pull request closed without merging**